.github/workflows/ci.yml

```diff
@@ -351,3 +362,24 @@ jobs:
               | tee -a $GITHUB_STEP_SUMMARY >&2
             exit 1
           fi
+
+  generate_cache:
+    runs-on: ubuntu-latest
+    name: Generate cache
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+          key: ${{ runner.os }}-cargo-test-CI2-${{ hashFiles('**/Cargo.toml') }}
+
+      - name: Cargo metadata
+        run: |
+          cargo metadata
+
+      - name: Clean build artifacts
+        run: |
+          cargo clean
```

Comment:
Since we don't cache `target` anymore, is running `cargo clean` necessary?

Reply:
It's not necessary now!

Reply:
I'd removing all instances of `cargo clean` as it is currently unnecessary.